Can't build Julia with Clang on Linux

```console
% make CC=clang CXX=clang++ 
    CC src/jltypes.o
/home/mose/repo/julia/src/jltypes.c:2002:13: error: initializer element is not a compile-time constant
            jl_typename_type,
            ^~~~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2027:43: error: initializer element is not a compile-time constant
    jl_typename_type->types = jl_svec(13, jl_symbol_type, jl_any_type /*jl_module_type*/,
                                          ^~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2047:44: error: initializer element is not a compile-time constant
    jl_methtable_type->types = jl_svec(12, jl_symbol_type, jl_any_type, jl_any_type,
                                           ^~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2092:52: error: initializer element is not a compile-time constant
                                        jl_svec(2, jl_any_type, jl_any_type),
                                                   ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2097:47: error: initializer element is not a compile-time constant
                                   jl_svec(3, jl_symbol_type, jl_any_type, jl_any_type),
                                              ^~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2102:51: error: initializer element is not a compile-time constant
                                       jl_svec(2, jl_tvar_type, jl_any_type),
                                                  ^~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2107:56: error: initializer element is not a compile-time constant
                                            jl_svec(2, jl_any_type, jl_any_type),
                                                       ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2110:45: error: initializer element is not a compile-time constant
    jl_svec_t *anytuple_params = jl_svec(1, jl_wrap_vararg((jl_value_t*)jl_any_type, (jl_value_t*)NULL));
                                 ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2122:61: error: initializer element is not a compile-time constant
    ((jl_datatype_t*)jl_type_type)->parameters = jl_svec(1, tttvar);
                                                            ^~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2161:52: error: initializer element is not a compile-time constant
                                        jl_svec(2, jl_long_type, jl_any_type),
                                                   ^~~~~~~~~~~~
./julia.h:1509:26: note: expanded from macro 'jl_long_type'
#define jl_long_type     jl_int64_type
                         ^~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2195:29: error: initializer element is not a compile-time constant
                            jl_any_type,
                            ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2218:29: error: initializer element is not a compile-time constant
                            jl_any_type, // Union{TypeMapEntry, Nothing}
                            ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2270:36: error: initializer element is not a compile-time constant
                        jl_svec(2, jl_symbol_type, jl_array_any_type),
                                   ^~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2283:36: error: initializer element is not a compile-time constant
                        jl_svec(2, jl_long_type, jl_type_union(symornothing, 2)),
                                   ^~~~~~~~~~~~
./julia.h:1509:26: note: expanded from macro 'jl_long_type'
#define jl_long_type     jl_int64_type
                         ^~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2289:36: error: initializer element is not a compile-time constant
                        jl_svec(5, jl_module_type, jl_any_type, jl_symbol_type, jl_long_type, jl_long_type),
                                   ^~~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2295:36: error: initializer element is not a compile-time constant
                        jl_svec(1, jl_long_type),
                                   ^~~~~~~~~~~~
./julia.h:1509:26: note: expanded from macro 'jl_long_type'
#define jl_long_type     jl_int64_type
                         ^~~~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2301:36: error: initializer element is not a compile-time constant
                        jl_svec(2, jl_any_type, jl_long_type),
                                   ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2307:36: error: initializer element is not a compile-time constant
                        jl_svec(1, jl_any_type),
                                   ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
/home/mose/repo/julia/src/jltypes.c:2313:36: error: initializer element is not a compile-time constant
                        jl_svec(2, jl_any_type, jl_any_type),
                                   ^~~~~~~~~~~
./julia_internal.h:438:41: note: expanded from macro 'jl_svec'
                n == sizeof((void *[]){ __VA_ARGS__ })/sizeof(void *),        \
                                        ^~~~~~~~~~~
/usr/include/sys/cdefs.h:459:53: note: expanded from macro '_Static_assert'
      [!!sizeof (struct { int __error_if_negative: (expr) ? 2 : -1; })]
                                                    ^~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[1]: *** [Makefile:217: jltypes.o] Error 1
make: *** [Makefile:76: julia-src-release] Error 2
% clang --version
clang version 13.0.0
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
The build is successful if removing the assertions at
https://github.com/JuliaLang/julia/blob/a2e4226d2e47a38ca4f15cfc69504b6e16631d79/src/julia_internal.h#L428-L430 https://github.com/JuliaLang/julia/blob/a2e4226d2e47a38ca4f15cfc69504b6e16631d79/src/julia_internal.h#L437-L439 https://github.com/JuliaLang/julia/blob/a2e4226d2e47a38ca4f15cfc69504b6e16631d79/src/julia_internal.h#L445-L447
but probably one would like to compile Julia without having to edit the source code.

--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -422,7 +422,7 @@ jl_svec_t *jl_perm_symsvec(size_t n, ...);
 
 // this sizeof(__VA_ARGS__) trick can't be computed until C11, but that only matters to Clang in some situations
 #if !defined(__clang_analyzer__) && !(defined(_COMPILER_ASAN_ENABLED_) || defined(_COMPILER_TSAN_ENABLED_))
-#ifdef __GNUC__
+#ifdef _COMPILER_GCC_
 #define jl_perm_symsvec(n, ...) \
     (jl_perm_symsvec)(__extension__({                                         \
             static_assert(                                                    \

